cmd/launch-taskrun/konflux/ecp_lookup.go

```diff
@@ -0,0 +1,104 @@
+package konflux
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ClientReader interface captures only the read operations we need for testability
+type ClientReader interface {
+	Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error
+	List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error
+}
+
+// findReleasePlan looks for a release plan applicable for a given application
+func FindReleasePlan(ctx context.Context, cli ClientReader, appName string, ns string) (ReleasePlan, error) {
+	var rp ReleasePlan
+
+	// Get all release plans in the namespace
+	planList := &ReleasePlanList{}
+	err := cli.List(ctx, planList, client.InNamespace(ns))
+	if err != nil {
+		return rp, fmt.Errorf("failed to lookup release plan in namespace %s: %w", ns, err)
+	}
+	if len(planList.Items) == 0 {
+		return rp, fmt.Errorf("no release plans found in namespace %s: %w", ns, err)
```

Comment:
`err` would be nil here so the wrapping is ineffectual.

Reply:
suggested change: 
```
return rp, fmt.Errorf("no release plans found in namespace %s", ns)
```

Reply:
Thanks, good catch!